lib/spack/spack/test/cmd/env.py

```diff
@@ -975,7 +977,7 @@ def create_v1_lockfile_dict(roots, all_specs):
         # Version one lockfiles use the dag hash without build deps as keys,
         # but they write out the full node dict (including build deps)
         "concrete_specs": dict(
-            (s.dag_hash(), s.to_node_dict(hash=ht.build_hash))
+            (s.dag_hash(), s.to_node_dict(hash=ht.dag_hash))
```

Comment:
Are we really creating a v1 lockfile? Shouldn't this read:
```suggestion
            (s.runtime_hash(), s.to_node_dict(hash=ht.build_hash))
```
? @tgamblin @becker33

Reply:
I've updated this test with these suggestions.